extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/FragmentMethodsAdder.java

```diff
@@ -34,12 +34,26 @@ public void add(ClassCreator classCreator, String generatedClassName,
             List<DotName> customInterfaceNamesToImplement, Map<String, FieldDescriptor> customImplNameToHandle) {
         for (DotName customInterfaceToImplement : customInterfaceNamesToImplement) {
             Collection<ClassInfo> knownImplementors = index.getAllKnownImplementors(customInterfaceToImplement);
-            if (knownImplementors.size() != 1) {
+            ClassInfo customInterfaceImpl = null;
+            if (knownImplementors.size() > 1) {
+                for (ClassInfo knownImplementor : knownImplementors) {
+                    if (knownImplementor.name().toString().endsWith("Impl")) { // the default suffix that Spring Data JPA looks for is 'Impl'
+                        if (customInterfaceImpl != null) { // make sure we don't have multiple implementations suffixed with 'Impl'
+                            throw new IllegalArgumentException(
+                                    "Interface " + customInterfaceToImplement
+                                            + " must contain a single implementation whose name ends with 'Impl'");
```

Comment:
This message should be improved to include the two conflicting implementations.

Reply:
Makes sense

Reply:
Fixed